source/contracts/reporting/Market.sol

```diff
@@ -127,555 +117,310 @@ contract Market is DelegationTarget, Extractable, ITyped, Initializable, Ownable
-        int256 _secondPlaceStake = getPayoutDistributionHashStake(bestGuessSecondPlaceTentativeWinningPayoutDistributionHash);
-        int256 _payoutStake = getPayoutDistributionHashStake(_payoutDistributionHash);
-
-        if (_tentativeWinningStake >= _secondPlaceStake && _secondPlaceStake >= _payoutStake) {
-            tentativeWinningPayoutDistributionHash = (_tentativeWinningStake > 0) ? tentativeWinningPayoutDistributionHash: bytes32(0);
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_secondPlaceStake > 0) ? bestGuessSecondPlaceTentativeWinningPayoutDistributionHash : bytes32(0);
-        } else if (_tentativeWinningStake >= _payoutStake && _payoutStake >= _secondPlaceStake) {
-            tentativeWinningPayoutDistributionHash = (_tentativeWinningStake > 0) ? tentativeWinningPayoutDistributionHash: bytes32(0);
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_payoutStake > 0) ? _payoutDistributionHash : bytes32(0);
-        } else if (_secondPlaceStake >= _tentativeWinningStake && _tentativeWinningStake >= _payoutStake) {
-            _payoutDistributionHash = tentativeWinningPayoutDistributionHash; // Reusing this as a temp value holder
-            tentativeWinningPayoutDistributionHash = (_secondPlaceStake > 0) ? bestGuessSecondPlaceTentativeWinningPayoutDistributionHash: bytes32(0);
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_tentativeWinningStake > 0) ? _payoutDistributionHash: bytes32(0);
-        } else if (_secondPlaceStake >= _payoutStake && _payoutStake >= _tentativeWinningStake) {
-            tentativeWinningPayoutDistributionHash = (_secondPlaceStake > 0) ? bestGuessSecondPlaceTentativeWinningPayoutDistributionHash: bytes32(0);
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_payoutStake > 0) ? _payoutDistributionHash: bytes32(0);
-        } else if (_payoutStake >= _tentativeWinningStake && _tentativeWinningStake >= _secondPlaceStake) {
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_tentativeWinningStake > 0) ? tentativeWinningPayoutDistributionHash: bytes32(0);
-            tentativeWinningPayoutDistributionHash = (_payoutStake > 0) ? _payoutDistributionHash: bytes32(0);
-        } else if (_payoutStake >= _secondPlaceStake && _secondPlaceStake >= _tentativeWinningStake) {
-            tentativeWinningPayoutDistributionHash = (_payoutStake > 0) ? _payoutDistributionHash: bytes32(0);
-            bestGuessSecondPlaceTentativeWinningPayoutDistributionHash = (_secondPlaceStake > 0) ? bestGuessSecondPlaceTentativeWinningPayoutDistributionHash: bytes32(0);
+    function finishedCrowdsourcingDisputeBond() public onlyInGoodTimes returns (bool) {
+        IReportingParticipant _reportingParticipant = IReportingParticipant(msg.sender);
+        require(isContainerForReportingParticipant(_reportingParticipant));
```

Comment:
It feels like we could make this callable by anyone and take in a `IReportingParticipant` parameter and `require(participant.isFull())`.  This should make auditing a bit easier as we don't need to rely on "the right caller".  Also, if some bug occurs that (for whatever reason) makes it so this method isn't called we can fix it by hand.

Reply:
Sounds reasonable. In general anytime we can remove these security checks we should since they cost an obscene amount of gas as well

Reply:
Starting implementing this but realized we still have to make sure the participant is legitimate, so really this just adds two new security checks (ensuring it is full and not already added) for no real benefit except potential hazards, so leaving it out unless there's a compelling reason to have it